spec/ruby/optional/capi/bignum_spec.rb

```diff
@@ -7,18 +7,29 @@ def ensure_bignum(n)
   n
 end
 
+full_range_longs = SpecGuard.new.implementation?(:jruby, :truffleruby)
+
 describe "CApiBignumSpecs" do
   before :each do
     @s = CApiBignumSpecs.new
-    @max_long = ensure_bignum(2**(0.size * 8 - 1) - 1)
-    @min_long = ensure_bignum(-@max_long - 1)
-    @max_ulong = ensure_bignum(2**(0.size * 8) - 1)
-  end
+
+    if full_range_longs
```

Comment:
Could this condition be done with `fixnum_max` and `fixnum_min`? If so we don't need to maintain an explicit list and duplicate logic.